crates/anvil-polkadot/tests/it/utils.rs

```diff
@@ -0,0 +1,149 @@
+use alloy_primitives::hex;
+use anvil_core::eth::EthRequest;
+use anvil_polkadot::{
+    api_server::{self, ApiHandle},
+    cmd::NodeArgs,
+    opts::Anvil,
+    spawn_anvil_tasks,
+    substrate_node::service::{self, Service},
+};
+use anvil_rpc::response::ResponseResult;
+use eyre::{Result, WrapErr};
+use futures::channel::oneshot;
+use parity_scale_codec::Decode;
+use polkadot_sdk::{
+    sc_cli::CliConfiguration,
+    sc_client_api::{BlockBackend, HeaderBackend},
+    sp_core::{storage::StorageKey, twox_128, H256},
+};
+use serde_json::{json, Value};
+
+pub struct TestNode {
+    pub service: Service,
+    pub api: ApiHandle,
+    pub _runtime_handle: tokio::runtime::Handle,
+}
+
+impl TestNode {
+    pub async fn new() -> Result<Self> {
+        let handle = tokio::runtime::Handle::current();
+
+        let mut anvil_args = Anvil {
+            global: foundry_cli::opts::GlobalArgs::default(),
+            node: NodeArgs::default(),
+            cmd: None,
+        };
+
+        anvil_args.node.no_mining = true;
+        anvil_args.node.mixed_mining = false;
+        anvil_args.node.port = 0; // auto-assign
+
+        let (anvil_config, mut substrate_config) = anvil_args.node.clone().into_node_config()?;
+        let anvil_config = anvil_config.set_silent(true);
+
+        let temp_dir = tempfile::tempdir()?;
+        let db_path = temp_dir.path().join("db");
+
+        substrate_config.set_base_path(Some(db_path));
+        let config = substrate_config.create_configuration(&anvil_args, handle.clone())?;
+        let service = service::new(&anvil_config, config)?;
+        let api = spawn_anvil_tasks(anvil_config, &service).await?;
+        tokio::task::yield_now().await;
```

Comment:
we could directly get a `AnvilNodeConfig` and `SubstrateNodeConfig` as params to this function.
`AnvilNodeConfig` already implements `Default`, as opposed to `NodeArgs`.
This would also enable us to modify this config for specific tests if needed.

We could then directly use the `spawn` function

Why is the yield_now needed?